File: dcp-solutions/2024-11-26/solution.py
```python
def merge_intervals(intervals):
    if not intervals:
        return []

    # Sort by start value
    intervals.sort(key=lambda x: x[0])

    merged = [intervals[0]]
    for current in intervals[1:]:
        last = merged[-1]
        if current[0] <= last[1]:
            merged.append((last[0], max(last[1], current[1])))
        else:
            merged.append(current)

    return merged
```

File: dcp-solutions/2024-11-26/solution.js
```javascript
function mergeIntervals(intervals) {
    intervals.sort((a, b) => a[0] - b[0]);

    const merged = [];
    let currentEnd = null;

    for (let start of intervals) {
        if (!currentEnd || start[0] > currentEnd) {
            if (currentEnd) {
                merged.push([currentEnd[0], currentEnd[1]]);
            }
            currentEnd = start;
        } else {
            currentEnd[1] = Math.max(currentEnd[1], start[1]);
        }
    }

    if (currentEnd) {
        merged.push([currentEnd[0], currentEnd[1]]);
    }

    return merged;
}
```